pages/price-feeds/use-real-time-data/starknet.mdx

```diff
@@ -0,0 +1,210 @@
+        eth_erc20_address: ContractAddress,
+        eth_usd_price_feed_id: u256,
+    ) {
+        self.pyth_address.write(pyth_address);
+        self.eth_erc20_address.write(eth_erc20_address);
+        self.eth_usd_price_feed_id.write(eth_usd_price_feed_id);
+    }
+
+    #[abi(embed_v0)]
+    impl FetchPrice of super::IFetchPrice<ContractState> {
+        fn example_method(
+            ref self: ContractState,
+            destination: ContractAddress,
+            amount_in_usd: u256,
+            pyth_price_update: ByteBuffer
+        ) {
+            let pyth = IPythDispatcher { contract_address: self.pyth_address.read() };
+            let eth_erc20 = IERC20CamelDispatcher {
+                contract_address: self.eth_erc20_address.read()
+            };
+            let caller = get_caller_address();
+            let contract = get_contract_address();
+
+            let pyth_fee = pyth.get_update_fee(pyth_price_update.clone(), eth_erc20.contract_address);
+            if !eth_erc20.transferFrom(caller, contract, pyth_fee) {
+                panic_with_felt252("insufficient allowance for fee");
+            }
+            if !eth_erc20.approve(pyth.contract_address, pyth_fee) {
+                panic_with_felt252("approve failed");
+            }
+
+            pyth.update_price_feeds(pyth_price_update);
+
+            // ETH/USD price feed ID
+            // The complete list of feed IDs is available at https://pyth.network/developers/price-feed-ids
+            let price = pyth
+                .get_price_no_older_than(self.eth_usd_price_feed_id.read(), MAX_PRICE_AGE)
+                .unwrap_with_felt252();
+            let price_u64: u64 = price.price.try_into().unwrap();
+        }
+    }
+}
+```
+
+<Callout type="info" emoji="ℹ️">
+  Unlike Ethereum, there is no native token on Starknet. You cannot pass tokens
+  implicitly when calling functions. Moreover, there is no concept of a
+  designated payer account, unlike Solana. In Starknet, all token transfers must
+  be performed explicitly by calling functions on the token's ERC20 contract. In
+  the case of the Pyth contract on Starknet, the caller must approve the fee
+  transfer before calling `update_price_feeds` or using similar methods. You can
+  use **STRK** or **ETH** to pay the fee, but STRK is preferred. The fee is
+  currently set to the minimum possible value (1e-18 STRK, 1 WEI).
+</Callout>
+
+The code snippet above does the following things:
+
+1. Call `pyth.get_update_fee` to get the fee required to update the Pyth price feeds.
+1. Call `pyth.update_price_feeds` and pass `pyth_price_update` to update the Pyth price feeds.
+1. Call `pyth.get_price_no_older_than` to read the price, providing the [price feed ID](https://pyth.network/developers/price-feed-ids) you wish to read.
+
+The code snippet below provides an example of how to update the Pyth price feeds on Starknet using the `pyth-starknet-js` in JavaScript:
+
+```ts {52} copy
```

Comment:
i don't love having this code snippet -- i thought we were trying to move a lot of this into the "Fetch Price Updates" guide. Maybe it would be better to have some shorter snippets showing just the starknet-specific code? e.g., how to get a ByteBuffer from a VAA? 

(In general the signal to noise ratio on this code snippet seems low, because most of the code is generic starknet stuff)